roxie/roxiemem/roxiemem.cpp

```diff
@@ -461,6 +537,69 @@ extern void releaseRoxieHeap()
     }
 }
 
+extern void notifyAllUnusedRoxieMem()
+{
+    char *blockStart = 0;
+    char *blockEnd = 0;
+
+    CriticalBlock b(heapBitCrit);
+
+    if (heapHugePagesPossible)
+    {
+        // release only large chunks
+        for (unsigned i = 0; i < heapBitmapSize; i++)
+        {
+            heap_t t = heapBitmap[i];
+            if (t==HEAP_ALLBITS)
+            {
+                if (blockStart == 0)
+                {
+                    blockStart = heapBase + (i * HEAP_BITS * HEAP_ALIGNMENT_SIZE);
+                    blockEnd = blockStart + (HEAP_BITS * HEAP_ALIGNMENT_SIZE);
+                }
+                else
+                    blockEnd += (HEAP_BITS * HEAP_ALIGNMENT_SIZE);
+            }
+            else if (blockStart > 0)
```

Comment:
picky:  Should be checking non null if (blockStart) rather than an ordered comparison.  (Technically I think it may be undefined behaviour e.g. What if the top bit of the pointer was set?)